### DISCUSSION TITLE
Redis transit encryption via TLS

### DISCUSSION THREAD

**Discussion by toriath** [shlinkio/shlink#1833] [Help wanted] [answered]
Hi,

we've tried to setup Shlink with a redis hosted on AWS Elasticache that had TLS enabled. We weren't able to get it to run until we disabled transit encryption on the redis cluster.

Is TLS with redis currently not supported or did we miss any required configuration for that?

**Comment by acelaya** [comment]
I'm not sure if anyone has tried this before, but I would imagine something needs to be explicitly passed to the redis abstraction for TLS to be properly handled, and Shlink is not doing it at the moment.

Are you passing anything different to Shlink when configuring redis? Using `tls://` as the servers schema perhaps?

**Reply by toriath** [reply]
We tried passing REDIS_SERVERS as `tls://<host>:<port>` but that did not work either. Shlink is then running into issues while booting which I guess is due to the db:create Lock not working.

How / Where does Shlink pass the redis urls to the abstraction layer?

**Reply by acelaya** [reply, marked as answer]
EDIT 2023-11-26: Shlink 3.7.0 now supports encrypted redis connections.

---

Thanks for checking.

Shlink uses its own abstraction, coming from https://github.com/shlinkio/shlink-common#redis-support.

This is then passed down to [predis](https://github.com/predis/predis), via [RedisFactory](https://github.com/shlinkio/shlink-common/blob/main/src/Cache/RedisFactory.php).

The environment variables are mapped down to config here https://github.com/shlinkio/shlink/blob/develop/config/autoload/cache.global.php

I quickly checked predis docs, and some parts made me think it could be enough with the tls schema, but that's clearly not the case.

I guess the RedisFactory will need some extra logic.

Can you create a new feature request in this repo? I will try to get it prioritized.

**Reply by toriath** [reply]
Thank you for explaining so thoroughly. 

I've opened #1834 for this, please let me know if I can provide anything else. Would've taken a look myself, but I'm not really fluent in PHP.
